File: bachend-development/nodejs/Todo-api-nodejs/utils/fileHelpers.js
const fs = require("fs");
const path = require("path");

function readData(callback) {
  fs.readFile("./notes.json", "utf-8", (err, data) => {
    if (err) callback(err, null);
    else callback(null, JSON.parse(data));
  });
}

function writeData(data, callback) {
  fs.writeFile("./notes.json", JSON.stringify(data, null, 4), callback);
}

function readStaticFile(file, callback) {
  const filePath = path.resolve(__dirname, "../public", file);
  fs.readFile(filePath, callback);
}

module.exports = { readData, writeData, readStaticFile };

File: bachend-development/django/book/book/books/file_helpers.py
import os
import json

def read_file(filename):
    file_path = os.path.join(os.path.dirname(__file__), filename)
    try:
        with open(file_path, 'r') as file:
            return json.load(file)
    except FileNotFoundError:
        return []
        

def write_file(fileName, data):
    with open(fileName, 'w') as file:
        json.dump(data, file);
        